task_03/js/main.js

```diff
@@ -199,6 +199,38 @@ console.log(prepend(10, prepend(20, null)));
 console.log(nth(arrayToList([10, 20, 30]), 1));
 // → 20
 //--------------------------------------------------------------
+// * Deep comparison *
+function deepEqual(obj1,obj2){
+  var result=true
+  for (var property in obj1) {
+    if (typeof(obj1[property])=="object" & obj1[property]!=null){
+        result = deepEqual(obj1[property],obj2[property]);
+    } else {   
+      if (obj1[property]!=obj2[property]){ 
+        result=false; 
+      }
+    }
+  }
+  for (var property in obj2) {
```

Comment:
@RaymondProduction You don't need to redefine **property** variable. It was already defined on line 205. You should always remember that there is no block-scope in JavaScript! But don't worry - you will learn more about **var** and **scope** in next lesson!